drivers/analog-io/max2201x/iio_max2201x.c

```diff
@@ -0,0 +1,919 @@
+	},
+	{
+		.name = "ldac",
+		.store = max2201x_iio_write_ao_config,
+		.priv = MAX2201X_IIO_LDAC,
+	},
+	{
+		.name = "ldac_available",
+		.show = max2201x_iio_read_enable_available,
+		.shared = IIO_SHARED_BY_ALL
+	},
+	END_ATTRIBUTES_ARRAY
+};
+
+static struct iio_device max2201x_iio_dev = {
+	.debug_reg_read = (int32_t (*)())max2201x_iio_reg_read,
+	.debug_reg_write = (int32_t (*)())max2201x_iio_reg_write,
+	.debug_attributes = max2201x_debug_attrs,
+};
+
+static int max2201x_iio_read_raw(void *dev, char *buf, uint32_t len,
+				 const struct iio_ch_info *channel,
+				 intptr_t priv)
+{
+	enum max2201x_channel ch = channel->address;
+	struct max2201x_iio_desc *desc = dev;
+	int32_t val;
+	int ret;
+
+	ret = max2201x_get_data(desc->max2201x_desc, ch, &val);
+	if (ret)
+		return ret;
+
+	return iio_format_value(buf, len, IIO_VAL_INT, 1, &val);
+}
+
+static int max2201x_iio_write_raw(void *dev, char *buf, uint32_t len,
+				  const struct iio_ch_info *channel,
+				  intptr_t priv)
+{
+	enum max2201x_channel ch = channel->address;
+	struct max2201x_iio_desc *desc = dev;
+	int32_t val;
+
+	iio_parse_value(buf, IIO_VAL_INT, &val, NULL);
+
+	return max2201x_set_data(desc->max2201x_desc, ch, val);
+}
+
+static int max2201x_iio_write_offset(void *dev, char *buf, uint32_t len,
+				     const struct iio_ch_info *channel,
+				     intptr_t priv)
+{
+	enum max2201x_channel ch = channel->address;
+	struct max2201x_iio_desc *desc = dev;
+	int16_t val;
+
+	iio_parse_value(buf, IIO_VAL_INT, (int32_t *)&val, NULL);
+
+	return max2201x_offset_calib(desc->max2201x_desc, ch, val);
+}
+
+static int max2201x_iio_write_gain(void *dev, char *buf, uint32_t len,
+				   const struct iio_ch_info *channel,
+				   intptr_t priv)
+{
+	enum max2201x_channel ch = channel->address;
+	struct max2201x_iio_desc *desc = dev;
+	uint16_t val;
+
+	iio_parse_value(buf, IIO_VAL_INT, (int32_t *)&val, NULL);
+
+	return max2201x_gain_calib(desc->max2201x_desc, ch, val);
+}
+
+static int max2201x_iio_write_slew_rate(void *dev, char *buf, uint32_t len,
+					const struct iio_ch_info *channel,
+					intptr_t priv)
+{
+	enum max2201x_channel ch = channel->address;
+	struct max2201x_iio_desc *desc = dev;
+	uint32_t val, i;
+	enum max2201x_sr_step_size step_size;
+	enum max2201x_sr_update_rate update_rate;
+	bool enable, hart;
+	int ret;
+
+	ret = max2201x_reg_read(desc->max2201x_desc,
+				ch ? MAX2201X_AO_SLEW_RATE_CH2 : MAX2201X_AO_SLEW_RATE_CH1, &val);
+	if (ret)
+		return ret;
+
+	step_size = no_os_field_get(MAX2201X_SR_STEP_SIZE_MASK, val);
+	update_rate = no_os_field_get(MAX2201X_SR_UPDATE_RATE_MASK, val);
+	enable = no_os_field_get(MAX2201X_SR_EN_MASK, val);
+	hart = no_os_field_get(MAX2201X_SR_SEL_MASK, val);
+
+	switch (priv) {
+	case MAX2201X_IIO_SLEW_RATE:
+		for (i = 0; i < NO_OS_ARRAY_SIZE(max2201x_enable_avail); i++)
+			if (!strcmp(buf, max2201x_enable_avail[i]))
+				break;
+
+		if (i == NO_OS_ARRAY_SIZE(max2201x_enable_avail))
+			return -EINVAL;
```

Comment:
it looks like this behavior is repeated a number of times in this switch case, can you create a function to call each type with the according array and array size?